src/components/InvoiceInput.tsx

```diff
@@ -82,14 +81,17 @@ const InvoiceInput = () => {
                 setLnurl("");
                 setInvoiceValid(true);
             }
+            setInvoiceError(undefined);
         } catch (e) {
             setInvoiceValid(false);
             setBolt12Offer(undefined);
             setLnurl("");
-            setInvoiceError(e.message);
             if (inputValue.length !== 0) {
                 input.setCustomValidity(t(e.message));
                 input.classList.add("invalid");
+                setInvoiceError(e.message);
+            } else {
+                setInvoiceError(undefined);
             }
```

Comment:
What if we add this before the `try` block?
```
        if (inputValue.length === 0) {
            input.setCustomValidity("");
            input.classList.remove("invalid");
            setInvoiceError(undefined);
            return;
        }
```
We could then simplify this part of the catch to this:
```suggestion
                input.setCustomValidity(t(e.message));
                input.classList.add("invalid");
                setInvoiceError(e.message);
```
This way we could avoid unnecessary validation and throwing errors for an empty input.

I guess we could also remove `input.setCustomValidity("");` and `input.classList.remove("invalid");` from the `try` block.